src/main/windows/editor.js

```diff
@@ -114,94 +149,294 @@ class EditorWindow extends BaseWindow {
+   * @returns {number[]}
+   */
+  getCandidatesScore (fileList) {
+    const { _openedFiles, _openedRootDirectory, id } = this
+    const buf = []
+    for (const pathname of fileList) {
+      let score = 0
+      if (_openedFiles.some(p => p === pathname)) {
+        score = -1
+      } else {
+        if (isChildOfDirectory(_openedRootDirectory, pathname)) {
+          score += 5
+        }
+        for (const item of _openedFiles) {
+          if (isChildOfDirectory(path.dirname(item), pathname)) {
+            score += 1
+          }
+        }
+      }
+      buf.push({ id, score })
+    }
+    return buf
   }
 
-  openFolder (pathname) {
-    if (this.quitting) return
+  reload () {
+    const { id, browserWindow } = this
+
+    // Close watchers
+    ipcMain.emit('watcher-unwatch-all-by-id', id)
+
+    // Reset saved state
+    this._directoryToOpen = ''
+    this._filesToOpen = []
+    this._markdownToOpen = []
+    this._openedRootDirectory = ''
+    this._openedFiles = []
+
+    browserWindow.webContents.once('did-finish-load', () => {
+      this.lifecycle = WindowLifecycle.READY
+      const { preferences } = this._accessor
+      const { sideBarVisibility, tabBarVisibility, sourceCodeModeEnabled } = preferences.getAll()
+      const lineEnding = preferences.getPreferedEOL()
+      browserWindow.webContents.send('mt::bootstrap-editor', {
+        addBlankTab: true,
+        markdownList: [],
+        lineEnding,
+        sideBarVisibility,
+        tabBarVisibility,
+        sourceCodeModeEnabled
+      })
+    })
 
-    const { browserWindow } = this
-    ipcMain.emit('watcher-watch-directory', browserWindow, pathname)
-    browserWindow.webContents.send('AGANI::open-project', pathname)
+    this.lifecycle = WindowLifecycle.LOADING
+    super.reload()
+  }
+
+  destroy () {
+    super.destroy()
+
+    // Watchers are freed from WindowManager.
+
+    this._directoryToOpen = null
+    this._filesToOpen = null
+    this._markdownToOpen = null
+    this._openedRootDirectory = null
+    this._openedFiles = null
+  }
+
+  get openedRootDirectory () {
+    return this._openedRootDirectory
   }
 
   // --- private ---------------------------------
 
-  // Only called once during window bootstrapping.
-  _openFile = async filePath => {
-    const { browserWindow } = this
-    const { menu: appMenu, preferences } = this._accessor
+  _getPreferredBackgroundColor (theme) {
+    // Hardcode the theme background color and show the window direct for the fastet window ready time.
+    // Later with custom themes we need the background color (e.g. from meta information) and wait
+    // that the window is loaded and then pass theme data to the renderer.
+    switch (theme) {
+      case 'dark':
+        return '#282828'
+      case 'material-dark':
+        return '#34393f'
+      case 'ulysses':
+      return '#f3f3f3'
+      case 'graphite':
+        return '#f7f7f7'
+      case 'one-dark':
+       return '#282c34'
```

Comment:
Pls add a `default:` case.

Reply:
The `default` case is the return statement or should I really add `default` case and remove the `return`?

Reply:
In fact, the current code should have no running problems. Adding a default case is just for code style or readability. So it depends on you.